Events.md

```diff
@@ -0,0 +1,225 @@
+
+It emits the `offerlist key`, the `maker` address, `offerId` and whether or not the user chose to `deprovision` the offer.
+
+By emitting this event an indexer knows whether or not an offer is live. And whether or not an offer is deprovisioned. This is important because we need to know this, when we try to calculate how much an offer locks in provision. See the description of `Credit` for more info.
+
+The `maker` is not needed for an indexer to work, but is needed for RPC calls, so it is emitted and indexed for that reason. The `offerlist key` is only indexed because it is needed for RPC calls.
+
+### OfferWrite
+
+This event is emitted when an offer is posted on Mangrove.
+
+It emits the `offerlist key`, the `maker` address, the `logprice`, the `gives`, the `gasprice`, `gasreq` and the offers `id`.
+
+By emitting the `offerlist key` and `id`, an indexer will be able to keep track of each offer, because offerlist and id together create a unique id for the offer. By emitting the `maker` address, we are able to keep track of who has posted what offer. The `logprice` and `gives`, enables an indexer to know exactly how much an offer is willing to give and at what price, this could for example be used to calculate a return. The `gasprice` and `gasreq`, enables an indexer to calculate how much provision is locked by the offer, see `Credit` for more information.
+
+The fields `offerlist key` and `maker` are indexed, so that we can filter on them when doing RPC calls.
+
+### CleanStart
+
+This event is emitted when a user tries to clean offers on Mangrove, using the build in clean functionality.
+
+It emits the `offerlist key`, the `taker` address and `offersToBeCleaned`, which is the number of offers that should be cleaned. By emitting this event, an indexer can save what `offerlist` the user is trying to clean and what `taker` is being used. This way it can keep a context for the following events being emitted (Just like `OrderStart`). The `offersToBeCleaned` is emitted so that an indexer can keep track of how many offers the user tried to clean. Combining this with the amount of `OfferFail` events emitted, then an indexer can know how many offers the user actually managed to clean. This could be used for analytics.
+
+The fields `offerlist key` and `taker` are indexed, so that we can filter on them when doing RPC calls.
+
+### CleanComplete
+
+This event is emitted when a Clean operation is completed.
+
+It does not emit any fields. This event is only needed in order to know that the clean operation is completed. This way an indexer can know exactly in what context we are in. It could emit the total bounty received, but in order to know who got the bounty, an indexer would still be needed or we would have to emit the taker address aswell, in order for an RPC call to find the data. But an indexer would still be able to find this info, buy collecting all the previous events. So we do not emit the bounty.
```

Comment:
```suggestion
It does not emit any fields. This event is only needed in order to know that the clean operation is completed. This way an indexer can know exactly in what context we are in. It could emit the total bounty received, but in order to know who got the bounty, an indexer would still be needed or we would have to emit the taker address as well, in order for an RPC call to find the data. But an indexer would still be able to find this info, by collecting all the previous events. So we do not emit the bounty.
```